src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -192,7 +193,7 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
         return context;
     }
 
-    private async Task<ImmutableArray<RazorVSInternalCodeAction>> GetDelegatedCodeActionsAsync(VersionedDocumentContext documentContext, RazorCodeActionContext context, Guid correlationId, IRazorLogger logger, CancellationToken cancellationToken)
+    private async Task<ImmutableArray<RazorVSInternalCodeAction>> GetDelegatedCodeActionsAsync(VersionedDocumentContext documentContext, RazorCodeActionContext context, Guid correlationId, ILogger logger, CancellationToken cancellationToken)
```

Comment:
Would be good to update endpoint classes to instead instantiate ILogger in the ctor (using the injected IRazorLoggerFactory). Then this ILogger would not be passed in via the method.

Reply:
Done: https://github.com/dotnet/razor/pull/9751